addons/mail/static/src/models/thread_cache/thread_cache.js

```diff
@@ -234,7 +234,11 @@ function factory(dependencies) {
             } else if (thread === this.env.messaging.moderation) {
                 return domain.concat([['need_moderation', '=', true]]);
             } else {
-                return domain.concat([['model', '=', thread.model], ['res_id', '=', thread.id]]);
+                return domain.concat([
+                    ['message_type', '!=', 'user_notification'],
+                    ['model', '=', thread.model],
+                    ['res_id', '=', thread.id],
+                ]);
```

Comment:
I still don't know about this fix. For me it is either incorrect or not sufficient.

I think we need to better decide what we want functionally for those type of messages before considering this PR.
I have actually no idea what we want right now.

- Do we want to simply not fetch them in this flux, only for chatter threads? (current solution)
- Do we still want to display them if those messages were known from other sources (eg. from the bus, in some situations from init messaging maybe)? (currently they will be shown)
- Why would we want such messages in other threads specifically (channels? inbox? moderation?)? Does it even make sense for them?

@alexkuhn @xdu-odoo please enlighten me

Reply:
The idea behind `user_notification` messages is that they must be notified to a specific user, while not being shown in message list.

I have a feeling this should not be a `mail.message` in the 1st place, and it has been done so because making a `mail.message` either sends email or move to inbox (intended behaviour), but we don't want other aspects like message in message list.

Basically python code sucks, and I'm not yet ready to tackle changes in python like this one. So I think it's acceptable to make small assumptions in JS like this one. I would definitely put a brief comment to explain why we do this, just to make it clear that's not nice but we are aware of that.

Reply:
> I have a feeling this should not be a `mail.message` in the 1st place

Well, not quite correct, but at least not a `mail.message` simply linked to thread as if it should be listed in message list. Depends how we think it should be cleanly designed, maybe python models should fine-tune link between thread and message so that a message could be linked to a thread without necessarily being in message list? Honestly don't have an answer for that, and kinda think it's hasty to put effort in thinking about this right now when we have more pressing (JS refactoring-related) issues...

Reply:
Added a comment